auraed/src/runtime/mod.rs

```diff
@@ -56,58 +57,88 @@ pub struct CoreService {}
-                            status,
-                            stdout: "-".to_string(),
-                            stderr: "-".to_string(),
-                            exit_code: "-".to_string(),
-                        };
-                        Ok(Response::new(response))
-                    }
+        // let rmeta = r.meta.expect("parsing request meta");
+
+        // Build command to execute.
+        let mut cmd =
+            command_from_string(&r.command).expect("command string parsing");
+
+        // Calculate the ID for the cell to run the command in
+        let cell_id = cell_name_from_string(&r.command)
+            .expect("cell name hash calculation");
+
+        info!("Spawning cell: {}", cell_id);
+
+        // Create the cgroup for the cell
+        let hierarchy = cgroups_rs::hierarchies::auto(); // v1/v2 cgroup switch automatically
+        let cgroup: Cgroup = CgroupBuilder::new(&cell_id)
+            .cpu()
+            .shares(10) // Use 10% of the CPU relative to other cgroups
+            .done()
+            .build(hierarchy);
+        // Attach the running command to the cgroup
+        let controller: &cgroups_rs::cpu::CpuController =
+            cgroup.controller_of().expect("cgroup controller");
+
+        // Spawn the command
+        let running = cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).spawn();
+        match running {
+            Ok(running) => {
+                let pid = running.id();
+                controller
+                    .add_task(&CgroupPid::from(pid as u64))
+                    .expect("attaching to cgroup");
+
+                // Wait for the command to terminate
+                let output = running.wait_with_output().expect("waiting process termination");
+
+                // Destroy the cgroup upon completion
+                // Note: https://github.com/kata-containers/cgroups-rs/issues/92
+                // Note: The library does not clean up the cgroup, so it needs to be
+                // Note: removed manually.
+                let controller_resp = controller.delete();
+                if controller_resp.is_err() {
+                    debug!("{:?}", controller_resp)
+                }
+                let cgroup_resp = cgroup.delete();
+                if cgroup_resp.is_err() {
+                    debug!("{:?}", cgroup_resp)
                 }
+
+                // Return the result synchronously
+                let meta = meta::AuraeMeta {
+                    name: r.command,
+                    message: "".to_string(),
+                };
+                let proc = meta::ProcessMeta { pid: pid as i32 };
+                let status = meta::Status::Complete as i32;
+
+                // Parse stdout from pipe
+                let stdout_val = String::from_utf8_lossy(&output.stdout);
+
+                // Parse stderr from pipe
+                let stderr_val = String::from_utf8_lossy(&output.stderr);
+
+                let response = ExecutableStatus {
```

Comment:
we only respond after the command has completed?  it would be nicer to respond before completion, perhaps with a streaming response for logging.